Terraform Crash

### Terraform Version
Terraform v0.9.2

### Affected Resource(s)
Please list the resources as a list, for example:
- aws_launch_configuration
- Used in conjunction with many others: Auto Scaling Group, EBS Volumes, ENI's, etc.

### Terraform Output
Here's the offending line where the crash occurred in the standard Terraform output:

```
module.mongos.aws_launch_configuration.mongodb (deposed #0): Destroying... (ID: josh-mong...d1249f3e)
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x48057ac]

goroutine 11800 [running]:
github.com/hashicorp/terraform/command.(*UiHook).PreApply(0xc4207624b0, 0xc420b440f0, 0xc42056dbd0, 0x0, 0x20, 0xc42089fe80, 0x0)
	/opt/gopath/src/github.com/hashicorp/terraform/command/hook_ui.go:65 +0xac
github.com/hashicorp/terraform/terraform.(*EvalApplyPre).Eval.func1(0x8082c40, 0xc4207624b0, 0xc420cbf490, 0xc420751800, 0xc42056dc0c)
	/opt/gopath/src/github.com/hashicorp/terraform/terraform/eval_apply.go:118 +0x4f
github.com/hashicorp/terraform/terraform.(*BuiltinEvalContext).Hook(0xc42016a1c0, 0xc42089fe80, 0xc420e57cc0, 0x32)
	/opt/gopath/src/github.com/hashicorp/terraform/terraform/eval_context_builtin.go:61 +0x6c
github.com/hashicorp/terraform/terraform.(*EvalApplyPre).Eval(0xc42089fc40, 0x80846c0, 0xc42016a1c0, 0x2, 0x2, 0xc42123e170, 0xb)
	/opt/gopath/src/github.com/hashicorp/terraform/terraform/eval_apply.go:119 +0xe0
github.com/hashicorp/terraform/terraform.EvalRaw(0x8051b80, 0xc42089fc40, 0x80846c0, 0xc42016a1c0, 0x0, 0x0, 0x0, 0x0)
	/opt/gopath/src/github.com/hashicorp/terraform/terraform/eval.go:53 +0x175
github.com/hashicorp/terraform/terraform.(*EvalSequence).Eval(0xc42089fcc0, 0x80846c0, 0xc42016a1c0, 0x2, 0x2, 0xc420d3dfc0, 0xb)
	/opt/gopath/src/github.com/hashicorp/terraform/terraform/eval_sequence.go:14 +0xc1
github.com/hashicorp/terraform/terraform.EvalRaw(0x8052440, 0xc42089fcc0, 0x80846c0, 0xc42016a1c0, 0x32, 0x0, 0x0, 0x32)
	/opt/gopath/src/github.com/hashicorp/terraform/terraform/eval.go:53 +0x175
github.com/hashicorp/terraform/terraform.(*EvalOpFilter).Eval(0xc420a2b7d0, 0x80846c0, 0xc42016a1c0, 0x2, 0x2, 0xc420d3dfa0, 0xb)
	/opt/gopath/src/github.com/hashicorp/terraform/terraform/eval_filter_operation.go:37 +0x4c
github.com/hashicorp/terraform/terraform.EvalRaw(0x8052200, 0xc420a2b7d0, 0x80846c0, 0xc42016a1c0, 0x0, 0x0, 0x0, 0x0)
	/opt/gopath/src/github.com/hashicorp/terraform/terraform/eval.go:53 +0x175
github.com/hashicorp/terraform/terraform.(*EvalSequence).Eval(0xc42089fac0, 0x80846c0, 0xc42016a1c0, 0x2, 0x2, 0xc420d3df30, 0xb)
	/opt/gopath/src/github.com/hashicorp/terraform/terraform/eval_sequence.go:14 +0xc1
github.com/hashicorp/terraform/terraform.EvalRaw(0x8052440, 0xc42089fac0, 0x80846c0, 0xc42016a1c0, 0x4f25880, 0xc420d3df08, 0x4c9a8a0, 0xc420d3df10)
	/opt/gopath/src/github.com/hashicorp/terraform/terraform/eval.go:53 +0x175
github.com/hashicorp/terraform/terraform.Eval(0x8052440, 0xc42089fac0, 0x80846c0, 0xc42016a1c0, 0xc42089fac0, 0x8052440, 0xc42089fac0, 0xc420797d40)
	/opt/gopath/src/github.com/hashicorp/terraform/terraform/eval.go:34 +0x4d
github.com/hashicorp/terraform/terraform.(*Graph).walk.func1(0x50e7ba0, 0xc420e57340, 0x0, 0x0)
	/opt/gopath/src/github.com/hashicorp/terraform/terraform/graph.go:126 +0xd4d
github.com/hashicorp/terraform/dag.(*Walker).walkVertex(0xc42034bb20, 0x50e7ba0, 0xc420e57340, 0xc420e57940)
	/opt/gopath/src/github.com/hashicorp/terraform/dag/walk.go:387 +0x392
created by github.com/hashicorp/terraform/dag.(*Walker).Update
	/opt/gopath/src/github.com/hashicorp/terraform/dag/walk.go:310 +0x9ca
```

### Panic Output
Crash.log: https://gist.github.com/josh-padnick/e92346c4256e7d3de81577c4e81b514f

### Expected Behavior
Terraform should have applied without issue.

### Actual Behavior
Crash!

### Steps to Reproduce
?

### Important Factoids
I'm doing a module development and was making a small number of changes. Terraform started wanting to re-create an Auto Scaling Group I'd already created so I manually deleted the Auto Scaling Group, ran `terraform apply`, still had issues, so then ran `terraform destroy`.

But before I ran `terraform destroy`, I noticed that one of my Launch Configurations was reported as being in the "deposed" state. And indeed that seems to have caused the crash somehow.

Re-running `terraform apply` then started giving me this error:

```
aws_autoscaling_group.mongodb: Error creating AutoScaling Group: AlreadyExists: AutoScalingGroup by this name already exists - A group with the name josh-mongod already exists
```

--- a/command/hook_count.go
+++ b/command/hook_count.go
@@ -42,6 +42,10 @@ func (h *CountHook) PreApply(
 	h.Lock()
 	defer h.Unlock()
 
+	if d.Empty() {
+		return terraform.HookActionContinue, nil
+	}
+
 	if h.pending == nil {
 		h.pending = make(map[string]countHookAction)
 	}

--- a/command/hook_ui.go
+++ b/command/hook_ui.go
@@ -59,6 +59,11 @@ func (h *UiHook) PreApply(
 	d *terraform.InstanceDiff) (terraform.HookAction, error) {
 	h.once.Do(h.init)
 
+	// if there's no diff, there's nothing to output
+	if d.Empty() {
+		return terraform.HookActionContinue, nil
+	}
+
 	id := n.HumanId()
 
 	op := uiResourceModify

